### ISSUE TITLE
Panic with a CompositeController webhook

### ISSUE THREAD

**Issue by orf** [metacontroller/metacontroller#383]
I'll try and narrow it down, but I'm seeing the following error when invoking a composite controller webhook:

```
metacontroller-0 metacontroller {"level":"error","ts":1634491814.0056887,"msg":"Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\ngoroutine 534 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic({0x142c360, 0x22f4af0})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/runtime/runtime.go:74 +0x85\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000704660})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/runtime/runtime.go:48 +0x75\npanic({0x142c360, 0x22f4af0})\n\t/usr/local/go/src/runtime/panic.go:1038 +0x215\nk8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetAPIVersion(...)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:218\nk8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GroupVersionKind(0x0)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:426 +0x64\nmetacontroller/pkg/controller/common.RelativeObjectMap.Insert(0xc00012fc00, {0x1810e18, 0xc000510018}, 0x0)\n\t/go/src/metacontroller/pkg/controller/common/common.go:192 +0x4e\nmetacontroller/pkg/controller/common.RelativeObjectMap.InsertAll(0xc000518000, {0x1810e18, 0xc000510018}, {0xc0003d4060, 0x4, 0xc0003bcf90})\n\t/go/src/metacontroller/pkg/controller/common/common.go:204 +0x45\nmetacontroller/pkg/controller/common.MakeRelativeObjectMap(...)\n\t/go/src/metacontroller/pkg/controller/common/common.go:291\nmetacontroller/pkg/controller/composite.(*parentController).syncParentObject(0xc000518000, 0xc000510018)\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:534 +0x405\nmetacontroller/pkg/controller/composite.(*parentController).sync(0xc000518000, {0xc000122960, 0xc000704660})\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:495 +0x37e\nmetacontroller/pkg/controller/composite.(*parentController).processNextWorkItem(0xc000518000)\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:298 +0xf3\nmetacontroller/pkg/controller/composite.(*parentController).worker(...)\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:287\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f67156c7290)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:155 +0x67\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x4616d9, {0x17a4ae0, 0xc0006d4510}, 0x1, 0xc00030b560)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:156 +0xb6\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0x10000011069a0, 0x3b9aca00, 0x0, 0x30, 0xc0000460ec)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:133 +0x89\nk8s.io/apimachinery/pkg/util/wait.Until(...)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:90\nmetacontroller/pkg/controller/composite.(*parentController).Start.func1.1()\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:263 +0x89\ncreated by metacontroller/pkg/controller/composite.(*parentController).Start.func1\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:261 +0x8f6\n","stacktrace":"k8s.io/apimachinery/pkg/util/runtime.HandleCrash\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/runtime/runtime.go:48\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:1038\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:221\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:735\nk8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetAPIVersion\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:218\nk8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GroupVersionKind\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:426\nmetacontroller/pkg/controller/common.RelativeObjectMap.Insert\n\t/go/src/metacontroller/pkg/controller/common/common.go:192\nmetacontroller/pkg/controller/common.RelativeObjectMap.InsertAll\n\t/go/src/metacontroller/pkg/controller/common/common.go:204\nmetacontroller/pkg/controller/common.MakeRelativeObjectMap\n\t/go/src/metacontroller/pkg/controller/common/common.go:291\nmetacontroller/pkg/controller/composite.(*parentController).syncParentObject\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:534\nmetacontroller/pkg/controller/composite.(*parentController).sync\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:495\nmetacontroller/pkg/controller/composite.(*parentController).processNextWorkItem\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:298\nmetacontroller/pkg/controller/composite.(*parentController).worker\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:287\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:90\nmetacontroller/pkg/controller/composite.(*parentController).Start.func1.1\n\t/go/src/metacontroller/pkg/controller/composite/controller.go:263"}
metacontroller-0 metacontroller panic: runtime error: invalid memory address or nil pointer dereference [recovered]
metacontroller-0 metacontroller 	panic: runtime error: invalid memory address or nil pointer dereference
metacontroller-0 metacontroller [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x93b364]
metacontroller-0 metacontroller
metacontroller-0 metacontroller goroutine 534 [running]:
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000704660})
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/runtime/runtime.go:55 +0xd8
metacontroller-0 metacontroller panic({0x142c360, 0x22f4af0})
metacontroller-0 metacontroller 	/usr/local/go/src/runtime/panic.go:1038 +0x215
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetAPIVersion(...)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:218
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GroupVersionKind(0x0)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/apis/meta/v1/unstructured/unstructured.go:426 +0x64
metacontroller-0 metacontroller metacontroller/pkg/controller/common.RelativeObjectMap.Insert(0xc00012fc00, {0x1810e18, 0xc000510018}, 0x0)
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/common/common.go:192 +0x4e
metacontroller-0 metacontroller metacontroller/pkg/controller/common.RelativeObjectMap.InsertAll(0xc000518000, {0x1810e18, 0xc000510018}, {0xc0003d4060, 0x4, 0xc0003bcf90})
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/common/common.go:204 +0x45
metacontroller-0 metacontroller metacontroller/pkg/controller/common.MakeRelativeObjectMap(...)
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/common/common.go:291
metacontroller-0 metacontroller metacontroller/pkg/controller/composite.(*parentController).syncParentObject(0xc000518000, 0xc000510018)
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:534 +0x405
metacontroller-0 metacontroller metacontroller/pkg/controller/composite.(*parentController).sync(0xc000518000, {0xc000122960, 0xc000704660})
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:495 +0x37e
metacontroller-0 metacontroller metacontroller/pkg/controller/composite.(*parentController).processNextWorkItem(0xc000518000)
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:298 +0xf3
metacontroller-0 metacontroller metacontroller/pkg/controller/composite.(*parentController).worker(...)
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:287
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f67156c7290)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:155 +0x67
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x4616d9, {0x17a4ae0, 0xc0006d4510}, 0x1, 0xc00030b560)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:156 +0xb6
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x10000011069a0, 0x3b9aca00, 0x0, 0x30, 0xc0000460ec)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:133 +0x89
metacontroller-0 metacontroller k8s.io/apimachinery/pkg/util/wait.Until(...)
metacontroller-0 metacontroller 	/go/pkg/mod/k8s.io/apimachinery@v0.22.1/pkg/util/wait/wait.go:90
metacontroller-0 metacontroller metacontroller/pkg/controller/composite.(*parentController).Start.func1.1()
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:263 +0x89
metacontroller-0 metacontroller created by metacontroller/pkg/controller/composite.(*parentController).Start.func1
metacontroller-0 metacontroller 	/go/src/metacontroller/pkg/controller/composite/controller.go:261 +0x8f6
```

**Comment by orf** [comment]
This happens if you return a `null` child:

```
{"children": [null], ...}
```

**Comment by grzesuav** [comment]
it related to serialization, why oyu are returning `null` as a child ?

**Comment by grzesuav** [comment]
I have a draft, just need to write some test to cover this case

**Comment by grzesuav** [comment]
Fixed and released in 2.1.1
